compiler/noirc_evaluator/src/acir/mod.rs

```diff
@@ -2692,13 +2697,11 @@ mod test {
     }
 
     #[test]
-    fn multiply_with_bool_should_not_emit_range_check() {
+    fn unchecked_mul_should_not_have_range_check() {
```

Comment:
I changed this test because part of this logic was moved to SSA. However, we can still check that unchecked multiplications don't produce a range check, and there's an SSA test that makes sure this particular case will transform that `mul` to `unchecked_mul`.